plugins/node/opentelemetry-instrumentation-pg/src/utils.ts

```diff
@@ -161,7 +169,35 @@ export function handleInvalidQuery(
   return result;
 }
 
+export function handleExecutionResult(
+  config: PgInstrumentationConfig,
+  span: Span,
+  pgResult: pgTypes.QueryResult | pgTypes.QueryArrayResult | unknown
+) {
+  if (
+    config.enhancedDatabaseReporting &&
+    config.responseHook !== undefined &&
+    pgResult !== undefined
+  ) {
+    safeExecuteInTheMiddle(
```

Comment:
`safeExecuteInTheMiddle` is just a wrapper around try-catch. Using try-catch is more readable, "simple" and requires fewer CPU cycles.

Reply:
I don't mind if you don't want to use `safeExecuteInTheMiddle` in your PR but if you want to stop using this util, please make a discussion for that on the core and don't force it on other PRs

Reply:
@Rauno56 @vmarchaud, indeed I wanted to stay aligned with what seemed to me like a convention of sorts.

Reply:
Maybe I'm missing something but is there an argument for using it, @vmarchaud? I didn't know there is a convention for using it nor have I collected that there is a convention against using it.

Reply:
> Maybe I'm missing something but is there an argument for using it, @vmarchaud?

For others i'll answer here too (we dicussed about this in private on slack ith rauno) but mainly for consistency and having a central util that handle try/catching in every instrumentation.
I think we'll stick with `safeExecuteInTheMiddle` here for now